frontend/src/components/note/NoteModal.vue

```diff
@@ -26,21 +26,50 @@ const emit = defineEmits(['note:submit']);
 // State
 const visible = defineModel<boolean>('visible');
 const formRef: Ref<InstanceType<typeof Form> | null> = ref(null);
+const showBringForward: Ref<boolean> = ref(false);
 
 // Default form values
 let initialFormValues: any = {
   createdAt: new Date(),
+  bringForwardDate: null,
+  bringForwardState: null,
   note: props.note?.note,
   noteType: NOTE_TYPES.GENERAL
 };
 
 // Form validation schema
 const formSchema = object({
+  bringForwardDate: mixed()
+    .nullable()
+    .when('noteType', {
+      is: (noteType: string) => noteType === NOTE_TYPES.BRING_FORWARD,
+      then: (schema) => schema.test('must select date', 'A date must be selected', (value) => value instanceof Date),
```

Comment:
Let's make the error message match the required error: `Bring forward date is a required field`